src/ros_tcp_endpoint/tcp_sender.py

```diff
@@ -53,29 +63,21 @@ def send_unity_message(self, topic, message):
             return
 
         serialized_message = ClientThread.serialize_message(topic, message)
-
-        try:
-            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            s.settimeout(self.timeout)
-            s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-            s.connect((self.unity_ip, self.unity_port))
-            s.sendall(serialized_message)
-            s.close()
-        except Exception as e:
-            rospy.loginfo("Exception {}".format(e))
+        self.queue.put(serialized_message)
 
     def send_unity_service(self, topic, service_class, request):
         if self.unity_ip == '':
             print("Can't send a message, no defined unity IP!".format(topic, request))
             return
 
         serialized_message = ClientThread.serialize_message(topic, request)
-
+        
         try:
             s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
             s.settimeout(self.timeout)
             s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
             s.connect((self.unity_ip, self.unity_port))
+            s.sendall(struct.pack('<f', self.timeout))
```

Comment:
Could this be sent via a Unity Service message? It feels like somewhat of an anti-pattern to pack undeclared header data into the start of the network stream.

Reply:
Point taken, I guess I can make it a proper message.